src/Materials/effect.ts

```diff
@@ -626,6 +651,30 @@ export class Effect implements IDisposable {
             return " " + attribute;
         }));
         Logger.Error("Defines:\r\n" + this.defines);
+        if (Effect.LogShaderCodeOnCompilationError) {
+            let pcw = this._pipelineContext as WebGLPipelineContext;

```

Comment:
We cannot type the context as webgl here as well (sorry :( )

It needs to be the root class so effect has no link with webgl

Reply:
As the shaders only exist in a WebGL context, I have added a type guard: let me know if it's ok for you.

I don't know if it disconnects Effect and WebGL enough, but I can see there are other references to WebGL in this class (`WebGLUniformLocation`, `Engine.webGLVersion` for eg).

Reply:
Ideally I would prefer something that all contexts will be force to provide (the uniforms are only used in the webGL context but your code could be executed if we fail compiling a webgpu shader)

Reply:
Ok, I have updated the `IPipelineContext` interface to be able to get the vertex/fragment code. I think it's what you meant?